src/BinOp.chpl

```diff
@@ -1044,4 +1045,585 @@ module BinOp
+            tmp = val / divideBy;
+            visted = true;
+          }
+          when "<<<" {
+            if !has_max_bits {
+              throw new Error("Must set max_bits to rotl");
+            }
+            // return (val << ra) | (val >> (max_bits - ra));
+            tmp = val << ra;
+            var botBits = makeDistArray(ra.size, bigint);
+            botBits = val;
+            if r.etype == int {
+              var shift_amt = max_bits - ra;
+              // cant just do botBits >>= shift_amt;
+              var divideBy = makeDistArray(ra.size, bigint);
+              divideBy = 1:bigint;
+              divideBy <<= shift_amt;
+              botBits = botBits / divideBy;
+              tmp += botBits;
+            }
+            else {
+              var shift_amt = max_bits:uint - ra;
+              botBits >>= shift_amt;
+              tmp += botBits;
+            }
+            visted = true;
+          }
+          when ">>>" {
+            if !has_max_bits {
+              throw new Error("Must set max_bits to rotr");
+            }
+            // return (val >> ra) | (val << (max_bits - ra));
+            tmp = val;
+            // cant just do tmp >>= ra;
+            var divideBy = makeDistArray(ra.size, bigint);
+            divideBy = 1:bigint;
+            divideBy <<= ra;
+            tmp = tmp / divideBy;
+
+            var topBits = makeDistArray(ra.size, bigint);
+            topBits = val;
+            if r.etype == int {
+              var shift_amt = max_bits - ra;
+              topBits <<= shift_amt;
+              tmp += topBits;
+            }
+            else {
+              var shift_amt = max_bits:uint - ra;
+              topBits <<= shift_amt;
+              tmp += topBits;
+            }
+            visted = true;
+          }
+        }
+      }
+      select op {
+        when "//" { // floordiv
+          [(ei,ri) in zip(tmp,ra)] ei = if ri != 0 then val/ri else 0:bigint;
+          visted = true;
+        }
+        when "%" { // modulo
+          // we only do in place mod when val != 0, tmp will be 0 in other locations
+          // we can't use ei = li % val because this can result in negatives
+          [(ei,ri) in zip(tmp,ra)] if ri != 0 then ei.mod(val, ri);
+          visted = true;
+        }
+        when "**" {
+          if || reduce(ra<0) {
+            throw new Error("Attempt to exponentiate base of type BigInt to negative exponent");
+          }
+          if has_max_bits {
+            [(ei,ri) in zip(tmp,ra)] ei.powMod(val, ri, max_size);
+          }
+          else {
+            tmp = val ** ra;
+          }
+          visted = true;
+        }
+      }
+    }
+    if (val.type == bigint && r.etype == bigint) ||
+       (val.type == bigint && (r.etype == int || r.etype == uint || r.etype == bool)) ||
+       (r.etype == bigint && (val.type == int || val.type == uint || val.type == bool)) {
+      // TODO we have to cast to bigint until #21290 is resolved
```

Comment:
Did we put in an Arkouda issue to update these once the chapel fix is released?

Reply:
Added #2007